api/src/main/java/com/google/sps/ValidateUsername.java

```diff
@@ -0,0 +1,50 @@
+package com.google.sps;
+
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.Query;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.cloud.FirestoreClient;
+import com.google.api.core.ApiFuture;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class ValidateUsername {
+  @PostMapping("/validateUsername")
+  String checkUsername(@RequestBody String username) throws Exception {
+    return isUniqueUsername(username);
+  }
+
+  private static String isUniqueUsername(String username) throws IOException {
+    Firebase.init();
+    Firestore db = FirestoreClient.getFirestore();
+
+    CollectionReference users = db.collection("users");
+
+    Query query = users.whereEqualTo("username", username);
+
+    try {
+      ApiFuture<QuerySnapshot> querySnapshot = query.get();
+      List<QueryDocumentSnapshot> queryDocuments = querySnapshot.get().getDocuments();
+
+      if (queryDocuments.isEmpty()) {
+        return "true";
+      }
+    } catch (Exception e) {
+      System.out.println(e);
```

Comment:
What's the advantage of printing this exception instead of letting it propagate upwards?

Reply:
Is it possible to do that without executing some code in a finally block? It seems cleaner to just handle it at this level.

Reply:
What if you just moved all that code outside of the try block?

Reply:
Unfortunately that doesn't work because one of the methods called throws an error, so it must be handled in a try/catch block.

Reply:
Wait hm I just searched up what an error was and apparently you're not supposed to try-catch them anyways? https://www.geeksforgeeks.org/errors-v-s-exceptions-in-java/

Reply:
Oops, I meant exception. The methods throw exceptions.

Reply:
Checked exceptions propagate by declaring `throws` in the method header https://www.geeksforgeeks.org/exception-propagation-java/

Reply:
Are you suggesting that I let the exception propagate beyond checkUsername()? Or handle it in checkUsername()?

Reply:
Hmm that's up to you, what do you think is the most reasonable behavior for this exception? My comment was originally just to point out that printing a caught exception is unnecessary, considering that letting it propagate to the highest level automatically does the same thing (prints a stack trace).

Reply:
Sounds good, fixed!

Reply:
Also just realized that there is a [difference between println and printStackTrace](https://stackoverflow.com/questions/12095378/difference-between-e-printstacktrace-and-system-out-printlne)